Make max attachment size configurable in reporting settings (#261844)
elastic/kibana: docs/reference/configuration-reference/reporting-settings.md

:   [How long](elasticsearch://reference/elasticsearch/rest-apis/api-conventions.md#time-units) each worker has to produce a report. If your machine is slow or under heavy load, you might need to increase this timeout. If a Reporting job execution goes over this time limit, the job is marked as a failure and no download will be available. Can be specified as a number of milliseconds. Defaults to `4m`.

## Email notification settings [reporting-email-settings]

`xpack.reporting.email.maxAttachmentSize`

> :   The maximum [byte size](elasticsearch://reference/elasticsearch/rest-apis/api-conventions.md#byte-units) of a report that can be sent as an email attachment for scheduled reports. Reports exceeding this size will not be attached, and a warning will be logged. Defaults to `10mb`.

Reviewer: ```suggestion
:   The maximum [byte size](elasticsearch://reference/elasticsearch/rest-apis/api-conventions.md#byte-units) of a report that can be sent as an email attachment for scheduled reports. Reports exceeding this size aren't attached, and a warning is logged. Defaults to `10mb`.
```
use present tense whenever possible
